instrumentation/http/src/main/java/brave/http/HttpServerRequest.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2013-2019 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.http;
+
+import brave.Span;
+import brave.internal.Nullable;
+import brave.propagation.Propagation;
+
+/**
+ * Marks an interface for use in {@link HttpServerHandler#handleReceive(HttpServerRequest)}. This
+ * gives a standard type to consider when parsing an incoming context.
+ *
+ * @since 5.7
+ */
+public abstract class HttpServerRequest {
+  static final Propagation.Getter<HttpServerRequest, String> GETTER =
```

Comment:
Now, we pretty much only need one GETTER for http (we'll still need specific types for RPC like gRPC and dubbo, as well GETTERs for messaging until the abstraction is done)